File: dialoglibrary/src/main/java/com/sdg/dialoglibrary/test/DialogManager.java
package com.sdg.dialoglibrary.test;

import android.content.Context;

import com.sdg.dialoglibrary.utils.SdgUtils;

public class DialogManager {

    private LoadingDialog loadingDialog;
    private ProgressDialog progressDialog;
    private static DialogManager dialogManager;

    public static DialogManager get() {
        if (dialogManager == null) {
            dialogManager = new DialogManager();
            return dialogManager;
        }
        return dialogManager;
    }

    public void showLoading(Context context){
        if(loadingDialog == null){
            loadingDialog = new LoadingDialog(context);
        }else {
            loadingDialog.dismiss();
        }
        loadingDialog.show();
    }

    public void hideLoading(){
        if(loadingDialog != null){
            loadingDialog.dismiss();
        }
    }

    public void showProgress(Context context){
        if(progressDialog == null){
            progressDialog = new ProgressDialog(context);
        }else {
            progressDialog.dismiss();
        }
        progressDialog.show();
    }

    public DialogManager setProgress(int progress){
        progressDialog.setProgress(progress);
        return this;
    }

    public void setCallBack(ProgressDialog.CallBack callBack){
        progressDialog.setCallBack(callBack);
    }

    public void hideProgress(){
        if(progressDialog != null){
            progressDialog.dismiss();
        }
    }
}

File: dialoglibrary/src/main/java/com/sdg/dialoglibrary/dialog/DialogManager.kt
package com.sdg.dialoglibrary.dialog

class DialogManager {
    private var mLoadingDialog: LoadingDialog? = null
    private var mProgressDialog: ProgressDialog? = null

    companion object {
        private var sInstance: DialogManager? = null
        fun get(): DialogManager {
            if (sInstance == null) {
                synchronized(DialogManager::class.java) {
                    if (sInstance == null) {
                        sInstance = DialogManager()
                    }
                }
            }
            return sInstance!!
        }
    }

    /**
     * show loading dialog
     */
    fun showLoading() {
        mLoadingDialog?.let {
            mLoadingDialog!!.dismiss()
        } ?: let {
            mLoadingDialog = LoadingDialog()
        }
        mLoadingDialog!!.show()
    }

    /**
     * dismiss loading dialog
     */
    fun hideLoading() {
        mLoadingDialog?.dismiss()
    }

    /**
     * @desc 进度条的dialog使用需要：先show，再set，最后hide
     * */
    fun showProgress(){
        mProgressDialog?.let {
            mProgressDialog!!.dismiss()
        } ?: let {
            mProgressDialog = ProgressDialog()
        }
        mProgressDialog!!.show()
    }

    /**
     * @desc 设置进度条 设置完在添加成功回调
     * @param progress
     * */
    fun setProgress(progress:Int) : DialogManager{
        mProgressDialog?.setProgress(progress)
        return this
    }

    fun setCallBack(callBack: ProgressDialog.ProgressCallBack){
        mProgressDialog?.setCallBack(callBack)
    }

    fun hideProgress(){
        mProgressDialog?.dismiss()
    }


}